File: Python/1_TwoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        :author: daiyu
        :time :2015-9-15
        :url: https://leetcode.com/problems/two-sum/
        """
        process = {}
        
        for index in range(len(nums)):
        #判断如果target － 当前值在字典中，那么返回这个结果值的value 与此时的Index值
            if target - nums[index] in process: 
                return [process[target-nums[index]] + 1, index + 1]
            #如果不在，则将该值与对应索引存入字典中
            process[nums[index]] = index

File: Java/1_TwoSum.java
/*
Input: numbers={2, 7, 11, 15}, target=9
Output: index1 = 1 index2 = 2
Url:https://leetcode.com/problems/two-sum/
Author:daiyu
Time:2015-9-15
Language:Java
*/
public Class Solution{
	public int[] twoSum(int[] nums, int target) {
		int[] result = new int[2];

		for (int i = 0; i < nums.length; i ++) {
			for (int j = i + 1; j < nums.length; j++) {
				if (nums[i] + nums[j] == target) {
					result[0] = i + 1;
					result[1] = j + 1;
					return result;
				}
			}
		}

		return result;
	}
}
